plugins/github/src/imbi/plugins/github/deployment.py

```
@@ -849,6 +875,225 @@ async def compare(
+        self, client: httpx.AsyncClient, commit_sha: str
+    ) -> dict[str, str | None]:
+        """Every note at one notes-ref commit, bodies included.
+
+        Blob reads run a few at a time (one request per note) and a
+        failed read skips its note rather than failing the batch or
+        recording a false "removed".
+        """
+        notes = await self._tree_notes(client, commit_sha)
+        gate = asyncio.Semaphore(_NOTE_BLOB_CONCURRENCY)
+
+        async def _read(blob_sha: str) -> str | None | BaseException:
```

Ruff (0.16.1)

[warning] `None` not at the end of the type union.

Move `None` to the end of the type union

(RUF036)